test/e2e/framework/providers/azure/azure.go

```diff
@@ -72,10 +72,14 @@ func (p *Provider) CreatePD(zone string) (string, error) {
 		PVCName:            pdName,
 		SizeGB:             1,
 		Tags:               nil,
-		AvailabilityZone:   zone,
 		DiskIOPSReadWrite:  "",
 		DiskMBpsReadWrite:  "",
 	}
+
+	// do not use blank zone definition
+	if len(zone) > 0 && zone != "0" {
```

Comment:
for the test, I think you could set like this:
```
if len(zone) > 0 {
 ...

```

Reply:
I would still like to explicitly not use zone "0" in createDisk calls because that is known to fail.

Reply:
I removed this check from here and we will rely on check in `CreateManagedDisk` function.

Reply:
thanks, I am also enabling the e2e test on availability zone cluster:
https://github.com/kubernetes/test-infra/pull/17436
let's wait for the e2e test result first.